resources/tools/mdsd/devops/pipeline/stages/impl/generic/modules/Metrics/JaCoCoQualityMetrics.groovy

```diff
@@ -7,4 +7,96 @@ jacoco([
     sourcePattern: '**/src,**/src-gen,**/xtend-gen',
     inclusionPattern: '**/*.class',
     exclusionPattern: '**/*Test*.class'
-])
\ No newline at end of file
+])
+
+// perform code coverage status check for pull requests
+if (CFG.isPullRequest) {
+
+    // constants
+    final COVERAGE_LEVEL = 'instructionCoverage'
+    final EPSILON = 0.001
+    final STATUS_CONTEXT = 'continuous-integration/jenkins/codecoverage'
+    final STATUS_URL = "${env.BUILD_URL}jacoco"
+    final CREDENTIALS_ID = '8adf889c-2157-45d1-acc7-1c6211538dac'
+
+    // pending notification
+    githubNotify credentialsId: CREDENTIALS_ID,
+                 context: STATUS_CONTEXT,
+                 targetUrl: STATUS_URL,
+                 status: 'PENDING',
+                 description: "Status check in progress..."
+
+    // search for master/main code coverage
+    echo "Collecting coverage of master branch"
+    def masterBranchName = 'main'
```

Comment:
Instead of checking against "main" and "master" we could actually go for querying github api for the default branch. But I guess this should be addressed more general, as currently several checks simply check against "master" and "main".

Reply:
I fully agree. I suggest to put this feature on hold by the end of the year as Github plans to add features that ease the transition. We might also switch completely to main, so this problem vanishes.